Simulator/Utils/RNG/Norm.cpp

```diff
@@ -56,31 +56,56 @@
 
 using namespace std;
 
+/*!
+   This method makes instances functors; it returns normally
+   distributed random numbers. Just a cute way of doing things.
+   @return pseudorandom number drawn from a normal distribution.
+*/
 BGFLOAT Norm::operator() ()
 {
-  BGFLOAT U1, U2,             // Uniformly distributed.
-    V1, V2, W, Y,            //Work variables (see above).
-    X1;                      // First value computed (returned immediately)
-
-  // Check to see if we need to compute anything, complement indicator.
-  if ((odd = !odd))
-    return(mu + sigma * X2);
-
-  // Do the computation step 1 (until W <= 1)
-  do {
-
-    U1 = MTRand::operator()();  /* Generate U(0,1) */
-    U2 = MTRand::operator()();
-    V1 = 2 * U1 - 1;
-    V2 = 2 * U2 - 1;
-    W = V1 * V1 + V2 * V2;
-  } while (W > 1);
-
-  // Do the computation step 2
-  Y = sqrt(-2 * log(W) / W);
-  X1 = V1 * Y;
-  X2 = V2 * Y;
-
-  // Return X1 this time, X2 next time
-  return(mu + sigma * X1);
+   BGFLOAT U1, U2,             // Uniformly distributed.
+      V1, V2, W, Y,            //Work variables (see above).
+      X1;                      // First value computed (returned immediately)
+
+   // Check to see if we need to compute anything, complement indicator.
+   if ((odd = !odd))
+      return(mu + sigma * X2);
+
+   // Do the computation step 1 (until W <= 1)
+   do {
+
+      U1 = MTRand::operator()();  /* Generate U(0,1) */
+      U2 = MTRand::operator()();
+      V1 = 2 * U1 - 1;
+      V2 = 2 * U2 - 1;
+      W = V1 * V1 + V2 * V2;
+   } while (W > 1);
+
+   // Do the computation step 2
+   Y = sqrt(-2 * log(W) / W);
+   X1 = V1 * Y;
+   X2 = V2 * Y;
+
+   // Return X1 this time, X2 next time
+   return(mu + sigma * X1);
 }
+
+Norm::~Norm() {}
+
+Norm::Norm(BGFLOAT m, BGFLOAT s, uint32_t seed)
+   : MTRand(seed), odd(true), mu(m), sigma(s) {}
+
+void Norm::seed(BGFLOAT m, BGFLOAT s, uint32_t seed) {
+   MTRand::seed(seed);
+   odd = true;
+   mu = m;
+   sigma = s;
+}
+
+void Norm::seed(uint32_t seed) {
+   Norm::seed(0.0, 1.0, seed);
+}
+
+void Norm::seed() {
+   Norm::seed(0.0, 1.0, 0);
+}
```

Comment:
Perhaps these could be made simpler and more consistent with default values for the method parameters?

Reply:
Unfortunately, I'm fairly certain that doing that for functions with the same function names would leave you with an infinite recursion.

Reply:
I you had `void Norm::seed(BGFLOAT m = 0.0, BGFLOAT s = 1.0, uint32_t seed = 0)`, then you wouldn't need a separate definition of `seed()`. Is there no way to avoid a conflict with `seed(uint32_t seed)`?

Reply:
I was fairly unclear ^, sorry about that. I can explain what I mean a little better when we meet.

`MTRand::seed()` and `MTRand::seed(uint32_t seed)` are both inherited by `Norm`. At least the second definition needs to be overridden to seed `MTRand *rgNormrnd` correctly.

`void Norm::seed(uint32_t seed = 0, BGFLOAT m = 0.0, BGFLOAT s = 1.0)` can't be used to override either of these.
`void Norm::seed(uint32_t seed = 0) override` compiles, but doesn't work as expected when `seed()` is called. It calls `MTRand::seed()`.

I would rather be more explicit with these definitions.
I'll specify the constant values using `#define`s, so that they're all specified in one place.

Reply:
OK, given that these need to be overridden, this is fine.